pumping/src/main/kotlin/com/dpm/pumping/auth/application/AuthService.kt

```diff
@@ -0,0 +1,52 @@
+package com.dpm.pumping.auth.application
+
+import com.dpm.pumping.auth.domain.LoginPlatform
+import com.dpm.pumping.auth.domain.LoginType
+import com.dpm.pumping.auth.dtos.AccessTokenResponse
+import com.dpm.pumping.auth.oauth2.dtos.OAuth2LoginResponse
+import com.dpm.pumping.auth.oauth2.dtos.SignUpRequest
+import com.dpm.pumping.user.domain.User
+import com.dpm.pumping.user.domain.UserRepository
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Transactional
+@Service
+class AuthService(
+    private val userRepository: UserRepository,
+    private val jwtTokenProvider: JwtTokenProvider,
+) {
+    fun login(appleUserId: String): OAuth2LoginResponse {
+
+        val user = userRepository.findByPlatform(LoginPlatform(LoginType.APPLE, appleUserId))
+            ?: run {
+                val platform = LoginPlatform(LoginType.APPLE, appleUserId)
+                userRepository.save(User.createWithOAuth(platform))
+                return OAuth2LoginResponse(null, null, platform.loginType, platform.oauth2Id)
+            }
+        val token = jwtTokenProvider.generateAccessToken(user.uid)
+        return OAuth2LoginResponse(
+            token.accessToken,
```

Comment:
정적 메서드 패턴 쓸거 아니면 `Named argument` 사용해서 빌더 패턴의 효과를 누리는건 어떠신지용
```suggestion
accessToken = token.accessToken,
```

Reply:
오우 이래서 빌더패턴이 필요없다는거군여 👍 반영했습니다~